conan/cli/commands/create.py

```diff
@@ -98,13 +98,20 @@ def create(conan_api, parser, *args):
     if test_conanfile_path:
         # TODO: We need arguments for:
         #  - decide update policy "--test_package_update"
-        tested_python_requires = ref.repr_notime() if is_python_require else None
+        # If it is a string, it will be injected always, if it is a RecipeReference, then it will
+        # be replaced only if ``python_requires = "tested_reference_str"``
+        tested_python_requires = ref.repr_notime() if is_python_require else ref
         from conan.cli.commands.test import run_test
         # The test_package do not make the "conan create" command return a different graph or
         # produce a different lockfile. The result is always the same, irrespective of test_package
         run_test(conan_api, test_conanfile_path, ref, profile_host, profile_build, remotes, lockfile,
                  update=False, build_modes=args.build, build_modes_test=args.build_test,
                  tested_python_requires=tested_python_requires, tested_graph=deps_graph)
+        if is_python_require:
+            if 'python_requires = "tested_reference_str"' not in load(test_conanfile_path):
```

Comment:
Only thing I dont like is the warning being conditionial on my linter deciding to add spaces between the equals and using double quotes. I'm guessing this might be the only way without actually parsing the file, so won't block :)

Reply:
Made it internal to ``run_test()`` so I can check the ``conanfile`` object directly instead, please re-check